src/install.ts

```diff
@@ -0,0 +1,35 @@
+import { App, reactive } from 'vue'
+import {
+  DriftWidget,
+  defaultConfig,
+  VueDriftWidgetConfig,
+  OptionalVueDriftWidgetConfig,
+} from './types'
+import { loadScript } from './utils'
+
+const VueDriftWidget: DriftWidget = {
+  install: (
+    app: App,
+    widgetId: string,
+    config: OptionalVueDriftWidgetConfig = defaultConfig,
+  ) => {
+    // Merge default and provided config
+    const driftConfig: VueDriftWidgetConfig = reactive({
+      ...defaultConfig,
+      ...config,
+    })
+
+    loadScript(widgetId, driftConfig)
+
+    app.provide('driftConfig', driftConfig)
+  },
+}
+
+export const createDrift = (
```

Comment:
after reviewing it now for several hours this field more like a plugin than a widget? it's a plugin creating the drift widget..  so maybe been renaming the repository to `vue-drift-widget-plugin` would make sense?

but this will be used as `app.use(createDrift, {...})`...but `use()` only takes two parameters... so I am a bit confused as `install` matches plugin usage.. but number of parameters does not... how is this full tool supposed to be used???

```suggestion
export const driftPlugin = (
```

Reply:
This will be used as 

```
createApp(App)
  .use(
    createDrift(import.meta.env.VITE_DRIFT_WIDGET_ID as string, {
      enabled: false,
      loadScript: false,
    }),
  )
  .mount('#app')
```

The config is a parameter of the createDrift function, not of the install function. That's btw how most of the plugins are doing it.

Reply:
see other comment about widget  being part of the configuration object.. why split it? you will not be able to just import the full config at once... e.g. `import driftConfig from './drift.config.ts'` or something like this... but this is not blocking just a decision that needs to be done at the beginning and thought through..

Reply:
changed in f55c995